index.js

```diff
@@ -359,15 +359,16 @@ exports.importPublic = function (publicKey) {
  * ECDSA sign
  * @param {Buffer} msgHash
  * @param {Buffer} privateKey
+ * @param {Number} [chainId]
  * @return {Object}
  */
-exports.ecsign = function (msgHash, privateKey) {
+exports.ecsign = function (msgHash, privateKey, chainId) {
   const sig = secp256k1.sign(msgHash, privateKey)
 
   const ret = {}
   ret.r = sig.signature.slice(0, 32)
   ret.s = sig.signature.slice(32, 64)
-  ret.v = sig.recovery + 27
+  ret.v = chainId ? sig.recovery + (chainId * 2 + 35) : sig.recovery + 27
```

Comment:
Ok, continuous operation of old/existing signature scheme is secured by making this optional.